Sources/KSCrashRecording/Monitors/KSCrashMonitor_NSException.m

```diff
@@ -135,8 +135,17 @@ static void setEnabled(bool isEnabled)
             
             KSLOG_DEBUG(@"Setting new handler.");
             NSSetUncaughtExceptionHandler(&handleUncaughtException);
-            KSCrash.sharedInstance.uncaughtExceptionHandler = &handleUncaughtException;
-            KSCrash.sharedInstance.currentSnapshotUserReportedExceptionHandler = &handleCurrentSnapshotUserReportedException;
+            
+            // --- WARNING ---
+            // This is in a `dispatch_async` as it's too dangerous,
+            // we're calling `KSCrash.sharedInstance` from a place
+            // that can easilly recurse into `KSCrash.sharedInstance`
+            // depending on the state of the system.
+            // We really shouldn't be doing this at all.
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Why is this done specifically on the main thread?

Reply:
We needed a serial queue to reduce the chances of a race condition. I also imagine most people `install` and `enable/disable` on the main queue. I thought it was a good choice.

I question if we actually need that code there to begin with. I don't know if it's used in the wild, we seem to only use it once from some example code.

Reply:
You know, I don't think it's related to OOM, so maybe it's better to think about it in a separate PR?

Reply:
I agree. What I noticed is that sometimes the tests fail without this fix. It's not always, but it does happen. Maybe it's only on my side and related to a change I made, I'm not sure.

Reply:
I'll remove it and figure out if there's a different root to the problem.

Reply:
I think the better solution for now is to disable that test and address it and this code in another PR

Reply:
I figured out the issue, but not the fix. I think mostly everyone uses the ObjC version of KSCrash which leads to the singleton being created very early. But if you don't do that and simply use kscrash_install, then when the exception monitor gets enabled the KScrash singleton is created for the first time and recurses into itself.

For now, a fix is simply to call `(void)KSCrash.sharedInstance` at the top of the test. ew.